crates/ide/src/syntax_highlighting/highlight.rs

```diff
@@ -703,6 +703,24 @@ fn highlight_name_ref_by_syntax(
     };
 
     match parent.kind() {
+        EXTERN_CRATE => {
+            let mut h: Highlight = HlTag::Symbol(SymbolKind::Module).into();
+            let is_crate_root = if let Some(extern_crate) = ast::ExternCrate::cast(parent.clone()) {
+                if let Some(first_segment) = extern_crate.name_ref() {
+                    first_segment.syntax().text() == name.syntax().text()
+                } else {
+                    false
+                }
+            } else {
+                false
+            };
+
+            if is_crate_root {
+                h |= HlMod::CrateRoot;
+            }
+
+            h | HlMod::Library
```

Comment:
```suggestion
            HlTag::Symbol(SymbolKind::Module) | HlMod::CrateRoot
```
this should suffice, it might not be a library (without semantic info we don't know), but its always going to be a crate root either way.